sycl/include/CL/sycl/detail/sycl_fe_intrins.hpp

```diff
@@ -49,3 +50,16 @@ extern "C" SYCL_EXTERNAL __attribute__((opencl_local)) std::uint8_t *
 __sycl_allocateLocalMemory(std::size_t Size, std::size_t Alignment);
 
 #endif
+
+namespace sycl {
```

Comment:
I am especially curious where this new type trait should live. I guessed at this location, but someone who does more in the SYCL runtime should suggest a better location and review whether I'm doing this "the SYCL way". This appears to be causing some SYCL runtime build failures, which suggests it's not quite right:
```
/localdisk2/sycl_ci/buildbot/worker/sycl-ubu-x64-pr/llvm.src/sycl/unittests/scheduler/CommandsWaitForEvents.cpp: In member function 'virtual void SchedulerTest_CommandsWaitForEvents_Test::TestBody()':
/localdisk2/sycl_ci/buildbot/worker/sycl-ubu-x64-pr/llvm.src/sycl/unittests/scheduler/CommandsWaitForEvents.cpp:94:3: error: reference to 'sycl' is ambiguous
   sycl::device HostDevice;
   ^~~~
```